### PR TITLE
Bazel build support

### PR DESCRIPTION
**Pull request by njlr**

### CODE REVIEW

**Comment by njlr** [comment]
@googlebot I signed it!

File: BUILD [hunk 1]
```diff
@@ -0,0 +1,35 @@
+cc_library(
```
Comment (ben-clayton) [inline BUILD]: Please add copyright headers to these new files

Comment (njlr) [reply BUILD]: Sure, can you point me to what I should copy-paste in?

Comment (ben-clayton) [reply BUILD]: See any other file, substituting the line comment characters if necessary 
Thanks!

File: BUILD [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  srcs = glob([
+    "src/**/*.cpp",
+    "src/**/*.c",
+    "src/**/*.S",
+    "src/**/*.h",
+  ], exclude = glob([
+    "src/**/*_test.cpp",
+  ])),
+  linkopts = select({
+    "@bazel_tools//src/conditions:linux_x86_64": [ "-pthread" ],
```
Comment (pmuetschard) [inline BUILD]: @ben-clayton what other platforms will need libpthread? I'm assuming at least the Android ones will need it as well.

**Comment by ben-clayton** [comment]
Looks good, thank you.
The CLA bot still seems grumpy. Are sure the form submit succeeded?

Cheers,
Ben

File: BUILD (hunk 2 above)
Comment (ben-clayton) [reply BUILD]: Likely, yes, but I'm happy to evolve the build rules than try to get them all perfect first time.

**Comment by njlr** [comment]
@googlebot I signed it!

**Comment by njlr** [comment]
> Looks good, thank you.
> The CLA bot still seems grumpy. Are sure the form submit succeeded?
> 
> Cheers,
> Ben

It is signed according to https://cla.developers.google.com/clas

**Comment by ben-clayton** [comment]
> It is signed according to https://cla.developers.google.com/clas

Odd. I've tried our internal checking tool and it doesn't seem to have an entry for your GitHub username. Is it possible that the email address you used to sign the CLA is different from the one associated with your GitHub account?

Cheers,
Ben

**Comment by njlr** [comment]
> > It is signed according to https://cla.developers.google.com/clas
> 
> Odd. I've tried our internal checking tool and it doesn't seem to have an entry for your GitHub username. Is it possible that the email address you used to sign the CLA is different from the one associated with your GitHub account?
> 
> Cheers,
> Ben

I changed my username at one point. I have updated the form with the new one to be safe. Perhaps it works now?

**Comment by ben-clayton** [comment]
It might need to be tickled with the:

@googlebot I signed it!

Magic line. Perhaps this message will suffice...

**Comment by ben-clayton** [comment]
Woo, that did the trick. Thanks!

File: BUILD [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 The Marl Authors.
```
Comment (ben-clayton) [inline BUILD]: Please can we rename this file to `BUILD.bazel`?
This collides with the default `build` output directory on case insensitive filesystems.

File: examples/BUILD [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2019 The Marl Authors.
```
Comment (ben-clayton) [inline examples/BUILD]: And this file to `BUILD.bazel` for consistency's sake.

**Comment by ben-clayton** [comment]
Perfect, thank you!

**Pull request merged**